### PR TITLE
Feature: Accounting core and VAT on purchase

### PR DESCRIPTION
**Pull request by tanmayjay**
#### Changes proposed in this Pull Request:
[new] Added sales return feature
[new] Added purchase return feature
[new] Added vat on purchase feature

### CODE REVIEW

File: modules/accounting/includes/functions/purchases.php [hunk 1]
```diff
@@ -639,6 +706,32 @@ function erp_acct_insert_purchase_data_into_ledger( $purchase_data ) {
             'updated_by'  => $purchase_data['updated_by'],
         ]
     );
+
+    if ( $purchase_data['tax'] ) {
+
+        $purchase_vat_ledger_id = $ledger_map->get_ledger_id_by_slug( 'purchase_vat' );
+        if ( ! $purchase_vat_ledger_id ) {
+            return new WP_Error( 505, 'Ledger ID not found for purchase vat', $purchase_data );
```
Comment (OpuRockey) [inline modules/accounting/includes/functions/purchases.php:714]: Error msg should be translatable. We need to take care of this.

File: includes/class-install.php [hunk 2]
```diff
@@ -823,10 +823,10 @@ public function create_tables() {
                 `currency` varchar(5) DEFAULT NULL,
                 `life_stage` VARCHAR(100) DEFAULT NULL,
                 `contact_owner` bigint(20) DEFAULT NULL,
-                `hash` VARCHAR(40) NULL DEFAULT NULL,
+                `hash` VARCHAR(40) DEFAULT NULL,
```
[the remark this answers is not in the document]
Comment (tanmayjay) [reply includes/class-install.php:826]: yes there is no issue in any version. in fact, this is the usual convention as any column is automatically nullable unless it is defined "NOT NULL".

File: includes/class-install.php [hunk 3]
```diff
@@ -935,8 +935,8 @@ public function create_tables() {
                 `status` varchar(25) DEFAULT NULL,
                 `subscribe_at` datetime DEFAULT NULL,
                 `unsubscribe_at` datetime DEFAULT NULL,
-                `hash` VARCHAR(40) NULL DEFAULT NULL,
-                PRIMARY KEY (`id`),
+                `hash` VARCHAR(40) DEFAULT NULL,
```
[the remark this answers is not in the document]
Comment (tanmayjay) [reply includes/class-install.php:938]: yes there is no issue in any version. in fact, this is the usual convention as any column is automatically nullable unless it is defined "NOT NULL".

File: includes/class-install.php [hunk 4]
```diff
@@ -1282,14 +1282,14 @@ public function create_tables() {
                 `trn_by` varchar(255) DEFAULT NULL,
                 `trn_by_ledger_id` int(11) DEFAULT NULL,
                 `particulars` varchar(255) DEFAULT NULL,
-                `ref` varchar(255) NULL DEFAULT NULL,
+                `ref` varchar(255) DEFAULT NULL,
```
[the remark this answers is not in the document]
Comment (tanmayjay) [reply includes/class-install.php:1285]: yes there is no issue in any version. in fact, this is the usual convention as any column is automatically nullable unless it is defined "NOT NULL".

File: includes/class-install.php [hunk 5]
```diff
@@ -1313,7 +1313,7 @@ public function create_tables() {
                 `amount` decimal(20,2) DEFAULT 0,
                 `trn_by` varchar(255) DEFAULT NULL,
                 `transaction_charge` decimal(20,2) DEFAULT 0,
-                `ref` varchar(255) NULL DEFAULT NULL,
+                `ref` varchar(255) DEFAULT NULL,
```
[the remark this answers is not in the document]
Comment (tanmayjay) [reply includes/class-install.php:1316]: yes there is no issue in any version. in fact, this is the usual convention as any column is automatically nullable unless it is defined "NOT NULL".

File: includes/class-install.php [hunk 6]
```diff
@@ -1445,6 +1445,8 @@ public function create_tables() {
                 `trn_date` date DEFAULT NULL,
                 `due_date` date DEFAULT NULL,
                 `amount` decimal(20,2) DEFAULT 0,
+                `tax` decimal(20,2) DEFAULT NULL ,
```
[the remark this answers is not in the document]
Comment (tanmayjay) [reply includes/class-install.php:1448]: Noted to be implemented in future.

**Comment by Rink9** [comment]
@tanmayjay 
Awaiting payment voucher issue.
Estimate voucher issue
Debit/credit inline

**Comment by Rink9** [comment]
@tanmayjay Balance sheet issue - when i edit any invoice ,data changes

**Pull request merged**